### DISCUSSION TITLE
Compare changes issues

### DISCUSSION THREAD

**Discussion by Gessy90** [community/community#49339] [Pull Requests]
### Select Topic Area

Question

### Body

![image](https://user-images.githubusercontent.com/42389173/223179773-3e5d5da6-29dd-4832-94b3-ce99df5e2c81.png)
Hi there,
Why I can't create a pull request here?

**Comment by airtower-luna** [comment]
You selected the same brach (`develop`) for "base" and "compare", so you're comparing the branch with itself. There'll never be a difference to compare or merge that way. As the message in the yellow box says, you need to choose two different branches. :wink: 

"Base" should be the branch you want to merge into, "compare" the one where your changes are.

**Comment by Gessy90** [comment]
Hi,
Thank you so much for your reply.
I don’t know why but when I search for the master branch it doesn’t appear in the list. Why is that?
Maybe I’m on the master branch already and that’s the reason why I can’t create a pull request. I’m really unsure.

Thanks for your reply

On 6 Mar 2023, at 18:45, Airtower ***@***.***> wrote:

﻿

You selected the same brach (develop) for "base" and "compare", so you're comparing the branch with itself. There'll never be a difference to compare or merge that way. As the message in the yellow box says, you need to choose two different branches. 😉

"Base" should be the branch you want to merge into, "compare" the one where your changes are.

**Comment by airtower-luna** [comment]
> I don’t know why but when I search for the master branch it doesn’t appear in the list. Why is that?

Maybe your repository doesn't have a `master` branch? It doesn't have to, that's just a common name for the default branch, increasingly being replaced with `main`. But you can name your branches as you like.

Possible reason: When you create a new, empty repository on GitHub, the first branch you push automatically becomes the default branch.

**Comment by Gessy90** [comment]
Ok, if this is the default brunch I don’t need to make a pull request, right?


On 6 Mar 2023, at 20:38, Airtower ***@***.***> wrote:

﻿

I don’t know why but when I search for the master branch it doesn’t appear in the list. Why is that?

Maybe your repository doesn't have a master branch? It doesn't have to, that's just a common name for the default branch, increasingly being replaced with main. But you can name your branches as you like.

Possible reason: When you create a new, empty repository on GitHub, the first branch you push automatically becomes the default branch.

**Comment by airtower-luna** [comment]
Most likely not, but it depends on how you want to manage your repository. :wink:

**Comment by airtower-luna** [comment]
If you have only one branch you literally *can't* make a PR. Most likely the best approach is to set this one to be the default branch (rename it if you wish) and then work with PRs for new code you want to add. But again, I don't know your repository and process. :slightly_smiling_face:
